### ISSUE TITLE
launch app directly with screensaver, not menu

### ISSUE THREAD

**Issue by JohnnyRico140** [theothernt/AerialViews#162]
I use AerialViews manually, launching it on my Android TV from home tv menu, and I'd like to know if I can have any random view directly, without getting into the menu first.

Thanks !

**Comment by theothernt** [comment]
That is something that has been suggested before, and I do hope to add it - I'm just not sure on a timeline just yet.

It maybe in the next few releases, but that would be a few months away.

**Comment by theothernt** [comment]
Just writing out some of my recent thoughts / ideas on this feature...

* Dreamy (removed from the Store) would show a settings icon if you used the d-pad. I can't use that as actions are already (or can be) assigned to directions, etc - could be confusing
* If app is launched, show message like "Use back button to access Settings" and launch into the TestActivity
* Any unhandled d-pad action, etc should exit/kill the app - customise with setting
* If screensaver is called by OS, no changes
* Not sure if this should be the default in future or not

**Comment by theothernt** [comment]
Pinging @wrybread

I hope to bring this feature - manually enabled - to v1.7.4

When enabled, the app launches into the screensaver directly - pressing back bring you to the regular app settings screen.

**Comment by wrybread** [comment]
I wonder about the logic of pressing Back to enter Settings. I think the user would expect pressing Back to exit the app, and Settings are pretty rarely accessed.

My way of solving the problem is to use an app that launches the screensaver, which I have set to Aerial Views. This app does that nicely:

https://play.google.com/store/apps/details?id=com.computerkim.screensavershortcut&hl=en_US

(There's also an APK version available, let me know if anyone needs me to find that).

Which makes me think that a better way might be to simply create another launcher icon/shortcut for Aerial Views that launches Aerial Views in video playback mode? So there's two shortcut icons: one that launches AV in Settings mode (like current behavior) and one that launches in video playback mode. I wonder if such a thing would be possible?

**Comment by theothernt** [comment]
> I wonder about the logic of pressing Back to enter Settings. I think the user would expect pressing Back to exit the app, and Settings are pretty rarely accessed.

That's a good point. I may allow the user to configure a button, maybe the OK button by default, but a button would have to be dedicated to opening the Settings screen.

> https://play.google.com/store/apps/details?id=com.computerkim.screensavershortcut&hl=en_US

It looks like this won't work if the Android TV 14 update comes out.

> Which makes me think that a better way might be to simply create another launcher icon/shortcut for Aerial Views that launches Aerial Views in video playback mode (even if it's not the screensaver)? So there's two shortcut icons: one that launches AV in Settings mode (like current behavior) and one that launches in video playback mode. I wonder if such a thing would be possible?

I can see some people being ok with that, but the other problem is the Play Store - not sure they'd allow an app with such limited functionality.

**Comment by wrybread** [comment]
> I may allow the user to configure a button, maybe the OK button by default, but a button would have to be dedicated to opening the Settings screen.

Good idea. One thing to keep in mind is that it's often not the power user who's using the TV, it's often someone else who expects a very simple experience, so I think it makes sense to make it possible for the main user to set up AV so people can use and exit the app without inadvertently making config changes, or even seeing the configs. 

> It looks like this won't work if the Android TV 14 update comes out.

That would be a shame. 

> the other problem is the Play Store

If it's an easy addition could maybe just make the companion app available for sideload only, if the Play Store won't allow it for whatever reason.

**Comment by theothernt** [comment]
> Good idea. One thing to keep in mind is that it's often not the power user who's using the TV, it's often someone else who expects a very simple experience, so I think it makes sense to make it possible for the main user to set up AV so people can use and exit the app without inadvertently making config changes, or even seeing the configs.

So changes to the app are ideally with a non-technical user in mind, ie. they should be able to install the app, run it... and it just works. With updates, I try not to change defaults very often and do my best to migrate settings forward.

With that in mind, the 'launch into screensaver on startup' option won't be a default - it'll be an advanced user feature as only a few people have ever asked for it.

> > It looks like this won't work if the Android TV 14 update comes out.
> That would be a shame.

I'll have to investigate at a later date but there is a difference between an app a) trying to launch another app vs b) trying to start the system screensaver.

**Comment by wrybread** [comment]
> I'll have to investigate at a later date but there is a difference between an app a) trying to launch another app vs b) trying to start the system screensaver.

Personally I don't think the companion app should try to launch the screensaver, I think it should launch AerialViews in video playback mode (regardless of whether it's registered as the screensaver or not).

**Comment by theothernt** [comment]
I'm targeting this feature for the next release, v1.7.5.

Be default, I'll map the "Exit to Setting" to a long-press of the OK button. But you can change that. The app will require "Exit to Settings" to be mapped to something... otherwise there will be no way to change any settings!

And the setting will be off by default *but* I may make it default in a future, bigger release - not sure yet.

**Comment by theothernt** [comment]
That feature is in the latest public beta if anyone wanted to test it?

**Comment by theothernt** [comment]
v1.7.5 has been released on GitHub, it should be out in the Play Store in the next week or so.
